packages/odyssey-react/src/components/Field/index.tsx

```diff
@@ -0,0 +1,169 @@
+   */
+  optionalLabel?: string;
+
+  /**
+   * Prefix to be read by SRs before announcing an error.
+   */
+  errorPrefix?: string;
+
+  /**
+   * the form field error
+   */
+  error?: string;
+
+  /**
+   * the form field hint
+   */
+  hint?: string;
+
+  /**
+   * The underlying input element required attribute
+   * @default true
+   */
+  required?: boolean;
+}
+
+export interface Props extends SharedFieldTypes {
+  /**
+   * Input to be rendered within the Field
+   */
+  children: ReactElement | ReactElement[];
+
+  /**
+   * The underlying input element id attribute. Automatically generated if not provided
+   */
+  inputId: string;
+
+  /**
+   * The underlying parent semantic HTML element.
+   * @default div
+   */
+  as?: "div" | "fieldset";
+}
+
+interface PropsLabel {
+  inputId: string;
+  optionalLabel?: string;
+  required: boolean;
+  children: ReactText;
+  as?: "label" | "legend";
+}
+interface PropsHint {
+  id: string;
+  children: ReactText;
+}
+interface PropsError {
+  id: string;
+  children: ReactText;
+  errorPrefix?: string;
+}
+
+export type StaticComponents = {
+  Label: typeof Label;
+  Hint: typeof Hint;
+  Error: typeof Error;
+};
+
+const Field: FunctionComponent<Props> & StaticComponents = (props) => {
+  const {
+    error,
+    hint,
+    inputId,
+    label,
+    optionalLabel,
+    errorPrefix,
+    required = true,
+    children,
+    as = "div",
+  } = props;
+
+  const Tag = as;
+  const TagLabel = as === "fieldset" ? "legend" : "label";
+
+  return (
+    <Tag className={styles.root}>
+      <Field.Label
+        inputId={inputId}
+        required={required}
+        optionalLabel={optionalLabel}
+        as={TagLabel}
+      >
+        {label}
+      </Field.Label>
+      {hint && <Field.Hint id={inputId}>{hint}</Field.Hint>}
+      {children}
+      {error && (
+        <Field.Error id={inputId} errorPrefix={errorPrefix}>
+          {error}
+        </Field.Error>
+      )}
+    </Tag>
+  );
+};
+
+const Label = (props: PropsLabel) => {
+  const {
+    inputId,
+    optionalLabel = "Optional",
+    required,
+    children,
+    as = "label",
+  } = props;
+
+  const Tag = as;
+
+  return (
+    <Tag
+      // We'll also need the ability to hide or reposition this label for UI like Search
+      className={styles.label}
+      htmlFor={inputId}
+    >
+      {children}
+      {!required && optionalLabel && (
+        <span className={styles.optionalLabel} children={optionalLabel} />
+      )}
+    </Tag>
+  );
+};
+
+const Hint = ({ id, children }: PropsHint) => (
+  <p className={styles.hint} id={`${id}-hint`} children={children} />
+);
+
+const Error = ({ id, children, errorPrefix = "Error:" }: PropsError) => (
+  <p className={styles.error} id={`${id}-error`}>
+    <ScreenReaderText>{errorPrefix}</ScreenReaderText>
+    {children}
+  </p>
+);
+
+Field.Label = Label;
+Field.Hint = Hint;
+Field.Error = Error;
+
+export default Field;
```

Comment:
we'll need to make sure any new components are using our HOC to apply their styles in test and production builds:

`export default withStyles(styles)(Field)`

Reply:
Typescript is asking you to provide the types for your statics here. These are the components you wrote in the PR here: `Field.Error` in this case. We can't hold off on this work as it represents a necessity to make our UI work at scale by handling styles and themes. It's a merge must have even if it requires some investigation and learning to make it happen. Let's work together if you need an assist.
[
Incidentally, my PR should solve this for you without the need to provide types for generics to `withStyles`](https://github.com/okta/odyssey/pull/1098) as I did for `List` previously to make this work as I'm describing above.